connector-runtime/connector-runtime-spring/src/main/java/io/camunda/connector/runtime/inbound/lifecycle/InboundConnectorManager.java

```diff
@@ -190,6 +193,33 @@ private void activateConnector(InboundConnectorDefinitionImpl newConnector) {
     }
   }
 
+  private void handlePollingConnector(
+      PollingConnectorExecutable executable,
+      InboundConnectorDefinitionImpl newConnector,
+      InboundConnectorContextImpl inboundContext) {
+    Function<Long, Map<String, String>> fetchVariablesFunction =
+        processDefinitionSearch::fetchVariablesByProcessInstanceKey;
+    Supplier<List<ProcessInstance>> processInstancesSupplier =
+        createProcessInstancesSupplier(fetchVariablesFunction, newConnector.processDefinitionKey());
+    ProcessInstanceContext processInstanceContext =
+        new ProcessInstanceContextImpl(processInstancesSupplier, new FeelEngineWrapper());
+    InboundIntermediateConnectorContextImpl intermediateContext =
+        new InboundIntermediateConnectorContextImpl(inboundContext, processInstanceContext);
```

Comment:
Please move this into a ContextFactory that creates the `InboundIntermediateConnectorContextImpl` instances.